sqlite-db/src/insert_rollover_completed_event_data.rs

```diff
@@ -0,0 +1,416 @@
+use anyhow::Context;
+use anyhow::Result;
+use bdk::bitcoin::hashes::hex::ToHex;
+use model::Cet;
+use model::CfdEvent;
+use model::Dlc;
+use model::EventKind;
+use model::FundingFee;
+use model::OrderId;
+use model::RevokedCommit;
+use sqlx::pool::PoolConnection;
+use sqlx::Connection as SqlxConnection;
+use sqlx::Sqlite;
+use sqlx::Transaction;
+
+pub async fn insert_rollover_completed_event(
+    connection: &mut PoolConnection<Sqlite>,
+    event_id: i64,
+    event: CfdEvent,
+) -> Result<()> {
+    let event_kind = event.event;
+    match event_kind {
+        EventKind::RolloverCompleted {
+            dlc: Some(dlc),
+            funding_fee,
+        } => {
+            // we begin a transaction so that it rollsback in case of a failure further down
+            let mut inner_transaction = connection.begin().await?;
+
+            delete_rollover_completed_event_data(&mut inner_transaction, event.id).await?;
```

Comment:
🛠 This is less resilient than having a separate task (i.e. `cull_old_dlcs`) that deals with the removal. 
If a rollover was completed it should be inserted no matter if deleting the old data worked or not. Otherwise we might fail on inserting an already finished rollover on deleting old data.

I think we can remove `cull_old_dlcs` as separate task, but I think we should still separate the deletion from inserting! i.e. the caller should first finish `insert_rollover_completed_event` and then call `delete_rollover_completed_event_data` (and if the delete errors we should just log the error but not report a failure).

Reply:
I disagree with this, the table is designed that there can only be **one** rollover completed event in the db. 
Having an extra actor which does this on a regular interval makes this unnecessary complex. 

I'm happy to make this infailible and just print an error if you prefer?

Reply:
To clarify: I don't think we need to have another error, but I think `insert` should not do the delete. 
I think the delete should be a separate call from the outside. I don't like making it infallible on the inside - I don't think it's a good API. What's the problem of having two calls on the outsides? Then the caller can decide what to do in case there was a problem.

Concerning 

> the table is designed that there can only be one rollover completed event in the db.

Does that mean you *have* to delete before you can insert?
I am not against it, but I feel that goes beyond the scope of this PR. This changes the constraint that we only remove data if we were able to save the new one. Doing this in a transaction mitigates that, but the problem that you might fail the insert on deleting remains. How would you make deleting infallible?

Reply:
I looked into it, imho this can't fail unless the db connection fails. Since this is all in a `transaction`, the whole `insert` should fail.

Reply:
How would it behave differently if you have two API functions outside?

Reply:
> How would you make deleting infallible?

The sql query is infailible, meaning, if there is nothing to delete, then nothing will be deleted :) This query should only fail if the connection to the db fails. 
Which, imho, is a severe error and we should probably try again? 
But this is something we would need to deal from the outside.

Reply:
Yeah, fair enough - this is an extension of our other discussion of update vs. delete + insert.
I think the update can be achieved by checking if a record exists. If not insert, if yes then update. But at this point I don't mind much. 
The only thing to note is, that the difference is that we will have a new ID in `rollover_completed_data` because we always insert. Otherwise we would literally have only one associated record where all the fields (besides the `id` and the `cfd_id`) would be updated. 
I don't think it makes much difference at the moment, but might make a difference later, depending on how this evolves. Happy to roll with the current implementation.